src/offset.rs

```diff
@@ -490,6 +490,125 @@ where
     }
 }
 
+/// An owned iterator over an offset
+pub struct OffsetIntoIter<T, OffsetItem: OffsetElement, Buffer: BufferType>
+where
+    T: IntoIterator,
+{
+    /// The underlying array data
+    data: <T as IntoIterator>::IntoIter,
+    /// The offsets buffer
+    offsets: <Buffer as BufferType>::Buffer<OffsetItem>,
```

Comment:
Can't we also use the `Buffer`'s `IntoIterator` instead of relying on indexing? You can wrap it in a [`Peekable`](https://doc.rust-lang.org/stable/std/iter/struct.Peekable.html). Then we don't need to keep `index`, but we just drain the `offsets` iterator.

Reply:
It looks much cleaner with `Peekable`, thanks for the suggestion!